### ISSUE TITLE
Sample application to build over the course of the book

### ISSUE THREAD

**Issue by jhliberty** [sane/sane-book#5]
@jamemackson brought up a good point: 

> One of the things I was thinking about while I was out of town was what “app” we’d build thru the 
> coarse of the book. @mgenev, is there a vision for what the how-to-sane app will be in the end?

And it seemed the general consensus was something that is **NOT** a todo list, blog or twitter clone and I would have to agree... Dead horses and all. 

Let's discuss some possible types of apps we could develop for the book here.

**Comment by jhliberty** [comment]
a couple suggestions that aren't too hard but aren't blogs, todo lists or twitter clones: 
- A Poll app, users can create polls and share them with friends, comment, and rate polls as well as get notified of the results when it ends.
- A dating site - users, profiles, geolocation data, messaging system, a simple matching algorithm, favorite and block lists
- A financial dashboard for stripe/braintree/simple that tracks payments and expenses

**Comment by mgenev** [comment]
The vision has been this -  to boil down the common denominator of features between most apps.

To me the obvious ones are:
- users, 
- social
- feed
- posts
- notifications
- private messaging
- vendors
- geo
- cms
- crm
- chat
- media gallery
- calendar
- store

all these and any others, in my opinion should be as generic as possible, staying generic serves 2 purposes: 
- it makes the project more useful, because it is reusable and can be built upon as a starting point
- makes it more feasible -  thinking of the 80/20 rule, in my experience 80% of the work is easy, it's the remaining 20% business-specific logic is what takes the majority of the time, we simply won't do the remaining 20% 

We do them as a learning project which matures, once it matures we turn it into an app making platform
akin to what wordpress is for CMS, but modern using Sane and Sane addons

I've been thinking of using this as a UI library for the how-to-sane demo, because it's free and generic and already has a lot of the features, but for every spin off app it can be replaced with the desired branding:
https://www.almsaeedstudio.com/preview

My idea has been to at some point fork how-to-sane into sane-startup 
- how-to-sane takes the path of an education platform which teaches how to make the features,
- sane-startup is the platform which allows a start up to make apps with the sane stack

I understand this is a lot of work, but enough people can get it done and my idea is that the book and the app, just keep growing anyway, not the typical book that gets written and dated in a year, if we do all this, we'll be looking at some of the most useful open source software out there and among the best 
way to learn how to make apps.

in a world where tech reigns supreme, this would be a very high standing :)

let me know what you think...

**Comment by mgenev** [comment]
poll, matching algorithm and financial dashboard are all things are i'm quite interested in as well and would be happy to work on

**Comment by mgenev** [comment]
staying generic serves 2 purposes: 
1) it makes the project more useful, because it can be built upon as a starting point
2) makes it more feasible -  thinking of the 80/20 rule, in my experience 80% of the work is easy, it's the remaining 20% business-specific logic is what takes the majority of the time, we simply won't do the remaining 20%

**Comment by jamemackson** [comment]
I love the vision @mgenev! I love the idea of the book continuing to evolve as the stack evolves.  And growing it into a real usable platform like that is a great side effect of all the efforts.  

I think the trick will obviously be in creating a phased approach to get there; ideally slicing up the functionality in such a way that it is conducive to the outline/progression of the book.  

If it would help the cause, I'd be happy at taking a stab at creating an expanded version of the book outline that would attempt to lay out functionality/components of the site that we would build in the context of the progression of chapters of the book.  I think having a more defined context might help us in dividing up the work of each chapter as well.  To that end, maybe instead of creating a separate outline, it would be better communicated to just go into each section of the book and add some "production notes" right in the body of the book.  Then we can use that as our guide when someone picks up a section to tackle.

It occurs to me also that, with the functional outline of the book as it is and the idea that we're then writing the content of the book in a style that progresses from one chapter to the next building up the site, the book will both function as a progressive tutorial and also be used as a reference guide for various functional challenges and other common patterns.  I think this will it an even more effective learning tool. (And I also really like the glossary features of gitbook for the ongoing reference side of this too)  I'm guessing that we'll be able to cover the basic and advanced with far less functionality than is in the big vision but maybe that's where we expand the book with a "common patterns" section that would cover adding these additional areas of the site.  I think this would be very useful to people too.

**Comment by jamemackson** [comment]
I also like the looks of that [UI library](https://www.almsaeedstudio.com/preview) too.  I also liked the library that someone (and I apologize I don't recall who it was) mentioned http://semantic-ui.com who actually ported some portions of their library to ember components https://github.com/Semantic-Org/Semantic-UI-Ember.  

I'm not sure if this is terribly unique or not (being the ember noob that I am) but it seems this extra step on their part might be something to consider if it is unique.

**Comment by jhliberty** [comment]
Good point @jamemackson - I'm a fan of semantic ui ember myself as well

**Comment by mgenev** [comment]
This is a good example of a similar tutorial we can learn from http://ember.vicramon.com/

**Comment by IanVS** [comment]
Looks great, but I'm nervous that there are no dates. Definitely need dates so people know how much to trust it.

**Comment by mgenev** [comment]
great point
